eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/serialization/core/Parameter.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.serialization.core;
+
+/**
+ * @author Gabor Keszthelyi
+ */
+public interface Parameter<T>
+{
+
+    void into(T container);
```

Comment:
I don't like this concept of mutating an existing container. Maybe consider using a builder style pattern like 

```
public interface Parametrized<T>
{
    T updated();

    Parametrized<T> withParameter(Key<V> key, V value);
}
```

Which would be used like this
```
Bundle newBundle = new Parametrized(oldBundle).withParameter(key, value).updated();
```

The builder should be immutable of course.

Not sure if that would work for all uses cases that you have in mind.

Also, the name `Parametrized` is not very good, because it would indicate a decorator, which is not the case. Maybe `BuildAdapter` would be more accurate (but it contains the pattern name, which is not very nice). I would call instances just like `BundleBuilder` or `IntentBuilder` omitting the "Adapter" part.